.github/workflows/helm-chart-ci.yaml

```
@@ -33,6 +33,28 @@ jobs:
       - name: Verify Docs updated
         run: ./helm-docs.sh
 
+  ensure-version-match:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.4.0
+      - name: Check objects for images without overrides
+        run: |
+          set +e
+          BASEVER=$(yq e .appVersion charts/spire/Chart.yaml)
+          for FILE in spiffe-oidc-discovery-provider spire-agent spire-server; do
+            VER=$(yq .appVersion charts/spire/charts/$FILE/Chart.yaml)
+            if [ "$VER" != "$BASEVER" ]; then
+              {
+                echo "## Version mismatch"
+                echo
+                echo "There is a mismatch between the chart versions (excluding the csi-driver). Please fix."
+              } >> "$GITHUB_STEP_SUMMARY"
+              exit 1
+            fi
+          done
```

```suggestion
      - name: Verify Spire appVersion
        run: |
          set +e
          BASEVER=$(yq e .appVersion charts/spire/Chart.yaml)
          for FILE in spiffe-oidc-discovery-provider spire-agent spire-server; do
            VER=$(yq .appVersion charts/spire/charts/$FILE/Chart.yaml)
            if [ "$VER" != "$BASEVER" ]; then
              {
                echo "## Version mismatch"
                echo
                echo "There is a mismatch between the chart versions (excluding the csi-driver and spire-controller-manager). Please fix."
              } >> "$GITHUB_STEP_SUMMARY"
              exit 1
            fi
          done
```

And rename the job name at line 26 to `checks` or something like that. Saves some build time.